tools/codegen/gen-config-mapper/provider_spec_model.go

```diff
@@ -0,0 +1,112 @@
+package genconfigmapper
+
+type ElemType int
+
+const (
+	Bool ElemType = iota
+	Float64
+	Int64
+	List
+	Map
+	Number
+	Object
+	Set
+	String
+)
+
+type CodeSpecification struct {
+	Resources Resource
+}
+
+type Resource struct {
+	Schema *Schema
+	Name   string
+}
+
+type Schema struct {
+	Description         *string
+	MarkdownDescription *string
+	DeprecationMessage  *string
```

Comment:
believe we can remove the markdown description for now and stick with description. Note that we have not been using the root schema description but might be useful to include.

Reply:
removed markdown description.

by root schema description, do you mean under Resource?

Reply:
meant under Schema, we can leave it for now 👍